### PR TITLE
Refactor bigquery scenario, include more than 100 rows

### PR DESCRIPTION
**Pull request by fejta**
/assign @cjwagner @rmmh 

* Make scenario pass `pylint`
* Make `--jq` argument optional
* Fix a variety of line breaks
* Require all files in metrics to be yaml files
* Make it faster to find the latest results by using a different name for each one.
* Make initial pydoc line less than 80 chars to match style guidelines.
* Include daily filtered result
* Put latest result in root
* Catch specific exceptions

### CODE REVIEW

File: scenarios/bigquery.py [hunk 1]
```diff
@@ -38,83 +40,96 @@ def check(*cmd, **keyargs):
     subprocess.check_call(*cmd, **keyargs)
 
 def validate_metric_name(name):
-    """Validates that a string is useable as a metric name (Can be used as GCS object name)."""
-    # Regex '$' symbol matches an optional terminating new line so we have to check that the name
+    """Raise if name is non-trivial."""
```
Comment (rmmh) [inline scenarios/bigquery.py]: `Raise ValueError`

File: scenarios/bigquery.py [hunk 2]
```diff
@@ -38,83 +40,96 @@ def check(*cmd, **keyargs):
...
     if not re.match(r'^[\w-]+$', name) or name[-1] == '\n':
-        raise ValueError('metric name must match the regular expression r\'^[\\w-]+$\'.')
+        raise ValueError(name)
 
 def do_query(query, out_filename, project):
     """Executes a bigquery query, outputting the results to a file."""
     with open(out_filename, 'w') as out_file:
-        check(['bq', 'query', '--format=prettyjson', '--project_id='+project], stdin=query,
-              stdout=out_file)
+        cmd = [
```
Comment (rmmh) [inline scenarios/bigquery.py]: gross

File: scenarios/bigquery_test.py [hunk 3]
```diff
@@ -32,35 +31,58 @@ class TestBigquery(unittest.TestCase):
...
-            with open(path) as config_file:
-                config = yaml.safe_load(config_file)
-                self.assertTrue(config)
-                self.assertTrue(config['metric'])
-                self.assertTrue(config['query'])
-                self.assertTrue(config['jqfilter'])
-                bigquery.validate_metric_name(config['metric'].strip())
+        """All yaml files in metrics are valid."""
+        start = os.path.join(os.path.dirname(__file__), '../metrics')
+        for root, _, names in os.walk(start):
```
Comment (rmmh) [inline scenarios/bigquery_test.py]: glob.glob is a lot easier to understand

File: scenarios/bigquery_test.py [hunk 4]
```diff
@@ -32,35 +31,58 @@ class TestBigquery(unittest.TestCase):
...
-        """Test that the do_jq function can execute a jq filter properly."""
+        """do_jq executes a jq filter properly."""
         # [filter, data, expected output]
-        tests = [['.', '{ "field": "value" }', '{"field":"value"}'],
-                 ['.field', '{ "field": "value" }', '"value"']]
+        tests = [
+            ['.', '{ "field": "value" }', '{"field":"value"}'],
+            ['.field', '{ "field": "value" }', '"value"'],
+        ]
         for test in tests:
```
Comment (rmmh) [inline scenarios/bigquery_test.py]: maybe `for expr, data, expected in tests`?

File: scenarios/bigquery.py [hunk 5]
```diff
@@ -38,83 +40,96 @@ def check(*cmd, **keyargs):
...
+        raise ValueError('bucket_path', bucket_path)
 
     # the 'bq show' command is called as a hack to dodge the config prompts that bq presents
     # the first time it is run. A newline is passed to stdin to skip the prompt for default project
     # when the service account in use has access to multiple projects.
-    check(['bq', 'show'], stdin='\n')
-
-    if not configs:
-        configs = [open(path) for path in glob.glob("test-infra/metrics/*.yaml")]
+    if nonsense:
```
Comment (rmmh) [inline scenarios/bigquery.py]: maybe replace with `if not os.path.exists(os.path.expanduser("~/.bigqueryrc")):`, which also helps hint at *why* we're executing this weird bit of code.

**Comment by rmmh** [comment]
/lgtm

social grooming applied, minor nits noted. no grubs detected

File: scenarios/bigquery.py (hunk 2 above)
Comment (fejta) [inline scenarios/bigquery.py]: Doesn't fit in 80 chars. Also added comment for why the `-n` flag is necessary

File: scenarios/bigquery.py (hunk 1 above)
Comment (fejta) [inline scenarios/bigquery.py]: Done

File: scenarios/bigquery.py (hunk 5 above)
Comment (fejta) [inline scenarios/bigquery.py]: Done

File: scenarios/bigquery_test.py (hunk 3 above)
Comment (fejta) [inline scenarios/bigquery_test.py]: Done, whose lame editor corrupts the working directory? vim certainly doesn't

File: scenarios/bigquery_test.py (hunk 4 above)
Comment (fejta) [inline scenarios/bigquery_test.py]: Done

**Comment by fejta** [comment]
The `-n10000` is the only thing strictly necessary here.

File: metrics/README.md [hunk 6]
```diff
@@ -6,22 +6,24 @@ database. The config files are consumed by the metrics-bigquery periodic prow jo
 ## Metrics
```
Comment (cjwagner) [inline metrics/README.md]: The first section of this README is no longer accurate, the naming conventions changed and there are now filtered results stored for each day.  It should be changed to something like:
> This folder contains config files describing metrics that summarize data in our Bigquery test result database. The config files are consumed by the metrics-bigquery periodic prow job that runs the bigquery image and scenario. Each metric consists of a bigquery query and a jq filter.  The query is run every 24 hours to produce a json data file containing the complete raw query results named with the format 'raw-yyyy-mm-dd.json'. The raw file is then filtered with the associated jq filter and the results are stored in 'daily-yyyy-mm-dd.json'. These files are stored in the k8s-metrics GCS bucket in a directory named with the metric name and persist for a year after their creation. Additionally, the latest filtered results for a metric are stored in the root of the k8s-metrics bucket and named with the format 'METRICNAME-latest.json'.

Comment (fejta) [inline metrics/README.md]: Done

**Pull request merged**